src/toplevelprops.ts

```diff
@@ -20,15 +21,45 @@ export interface TopLevelProperties {
   padding?: Padding;
 
   /**
-   * Resize is a boolean indicating if autosize layout should be re-calculated on every update.
+   * Sets how the visualization size should be determined. If a string, should be one of `"pad"` (default), `"fit"` or `"none"`.
+   * Object values can additionally specify parameters for content sizing and automatic resizing. See the autosize section below for more.
+   *
+   * __Default value__: `pad`
+   */
+  autosize?: AutosizeType | Autosize;
+}
+
+export type AutosizeType = 'pad' | 'fit' | 'none';
+
+export interface Autosize {
```

Comment:
Name this `AutoSizeParams`?  

Technically, `Autosize = AutosizeType | AutoSizeParams`.  
(This will be consistent with `bin: boolean: BinParams`, etc.)

Reply:
Done